src/test/java/org/jabref/logic/importer/CffImporterTest.java

```diff
@@ -0,0 +1,72 @@
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.fileformat.CffImporter;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CffImporterTest {
+
+    @Test
+    public void PreferredCitationTest() throws Exception {
+        String cffContent = """
+                cff-version: 1.2.0
+                message: "If you use this software, please cite it as below."
+                authors:
+                - family-names: "Lisa"
+                  given-names: "Mona"
+                  orcid: "https://orcid.org/0000-0000-0000-0000"
+                - family-names: "Bot"
+                  given-names: "Hew"
+                  orcid: "https://orcid.org/0000-0000-0000-0000"
+                title: "My Research Software"
+                version: 2.0.4
+                doi: 10.5281/zenodo.1234
+                date-released: 2017-12-18
+                url: "https://github.com/github-linguist/linguist"
+                preferred-citation:
+                  type: article
+                  authors:
+                  - family-names: "Lisa"
+                    given-names: "Mona"
+                    orcid: "https://orcid.org/0000-0000-0000-0000"
+                  - family-names: "Bot"
+                    given-names: "Hew"
+                    orcid: "https://orcid.org/0000-0000-0000-0000"
+                  doi: "10.0000/00000"
+                  journal: "Journal Title"
+                  month: 9
+                  start: 1
+                  end: 10
+                  title: "My awesome research software"
+                  issue: 1
+                  volume: 1
+                  year: 2021
+                """;
+
+        CffImporter importer = new CffImporter();
+        try (BufferedReader reader = new BufferedReader(new StringReader(cffContent))) {
+            ParserResult result = importer.importDatabase(reader);
+            List<BibEntry> entries = result.getDatabase().getEntries();
+
+            assertEquals(1, entries.size());
+            BibEntry entry = entries.get(0);
+
+            assertEquals("My awesome research software", entry.getField(StandardField.TITLE).orElse(null));
+            assertEquals("Mona Lisa and Hew Bot", entry.getField(StandardField.AUTHOR).orElse(null));
+            assertEquals("10.0000/00000", entry.getField(StandardField.DOI).orElse(null));
+            assertEquals("Journal Title", entry.getField(StandardField.JOURNAL).orElse(null));
+            assertEquals("1", entry.getField(StandardField.VOLUME).orElse(null));
+            assertEquals("1", entry.getField(StandardField.ISSUE).orElse(null));
+            assertEquals("1-10", entry.getField(StandardField.PAGES).orElse(null));
+            assertEquals("2021", entry.getField(StandardField.YEAR).orElse(null));
+            System.out.println("Test Successful!");
```

Comment:
Please rewrite using the idea of org.jabref.logic.database.DatabaseMergerTest#mergeAddsNonDuplicateEntries: List.of, BibEntry.withfield...

Please no Println with "test successful". IntelliJ will have green or red indicators. See the last part (marked as "optional") in our guide: https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.html

Reply:
Sure , I'll make the necessary changes.